emily/handler/src/database/entries/mod.rs

```diff
@@ -1,8 +1,435 @@
+            .set_item(Some(entry_item.into()))
+            .send()
+            .await?;
+        // Return.
+        Ok(())
+    }
+
+    /// Get all entries from a dynamodb table.
+    async fn get_all_entries(
+        dynamodb_client: &aws_sdk_dynamodb::Client,
+        settings: &Settings,
+    ) -> Result<Vec<Self::Entry>, Error> {
+        // Get table name.
+        let table_name = Self::table_name(settings);
+        // Create vector to aggregate items in.
+        let mut all_entries: Vec<Self::Entry> = Vec::new();
+        // Scan the table for as many entries as possible.
+        let mut scan_output = dynamodb_client.scan().table_name(table_name).send().await?;
+        // Put items into aggregate list.
+        all_entries.extend(serde_dynamo::from_items(
+            scan_output.items.unwrap_or_default(),
+        )?);
+        // Continue to query until the scan is done.
+        while let Some(exclusive_start_key) = scan_output.last_evaluated_key {
+            scan_output = dynamodb_client
+                .scan()
+                .table_name(table_name)
+                .set_exclusive_start_key(Some(exclusive_start_key))
+                .send()
+                .await?;
+            all_entries.extend(serde_dynamo::from_items(
+                scan_output.items.unwrap_or_default(),
+            )?);
+        }
+        // Return.
+        Ok(all_entries)
+    }
+
+    /// Generic delete table entry.
+    async fn delete_entry(
+        dynamodb_client: &aws_sdk_dynamodb::Client,
+        settings: &Settings,
+        key: &<Self::Entry as EntryTrait>::Key,
+    ) -> Result<(), Error> {
+        // Get table name.
+        let table_name = Self::table_name(settings);
+        // Convert Entry into the type needed for querying.
+        let key_item: Item = serde_dynamo::to_item(key)?;
+        // Add to the database.'
+        dynamodb_client
+            .delete_item()
+            .table_name(table_name)
+            .set_key(Some(key_item.into()))
+            .send()
+            .await?;
+        // Return.
+        Ok(())
+    }
+
+    /// Deletes every entry in a table with the specified keys.
+    async fn delete_entries(
+        dynamodb_client: &aws_sdk_dynamodb::Client,
+        settings: &Settings,
+        keys_to_delete: Vec<<Self::Entry as EntryTrait>::Key>,
+    ) -> Result<(), Error> {
+        // Get table name.
+        let table_name = Self::table_name(settings);
+        // Make delete request aggregator.
+        let mut write_delete_requests: Vec<WriteRequest> = Vec::new();
+        for key in keys_to_delete {
+            let key_item = serde_dynamo::to_item::<<Self::Entry as EntryTrait>::Key, Item>(key)?;
+            let write_request = WriteRequest::builder()
+                .delete_request(
+                    DeleteRequest::builder()
+                        .set_key(Some(key_item.into()))
+                        .build()?,
+                )
+                .build();
```

Comment:
nit: not bad, but I find it a little easier to read this when laid out like so:
```rust
let req = DeleteRequest::builder()
    .set_key(Some(key_item.into()))
    .build()?;
let write_request = WriteRequest::builder().delete_request(req).build();
```

Reply:
Done!